### DISCUSSION TITLE
Why not set the graph tab as the default section on DAG detail page?

### DISCUSSION THREAD

**Discussion by Liaomessi** [apache/airflow#27652] [Q&A] [answered]
DAG graph pages are great, but neither original tree view nor grid tables have a great user experience. So why not set the graph tab as the default section on DAG detail page?

**Comment by potiuk** [comment, marked as answer]
Could you be specific about "not greate user experience"? I think if you refer to "seeing nice graph", then yeah likely grid and tree view (TreeView is already gone BTW) do not have nice graphs, but they are immensly usable for the primary users of Airlfow - people who are managing Airlfow instance.

But maybe you have another criteria for what usability means. Could you please elaborate what you mean by "great user experience" and why you think graph view is better.

I am genuinly interested to hear more about - it - especially if you can destill it down to objective reasons and including the main audience of Airlfow as "personas" - because this is how usually you should talk about the user experience. You imagine Personas who are using the software and you explain how the user experience maps to what they can do.

I think it's good if such analysis does not have any "I like this more and that more", "this is nicer to me" - because often when you are judging the experience, you look only at your own specific, very narrow and limited needs. It should be instead put in a much broader context where multiple perspectives and user stories are taken into account.

I would love to hear more in that direction. Would you like to elaborate on it in this direction?

**Reply by Taragolis** [reply]
In additional users always have the option to setup default view for all DAG on webserver by set configuration [[webserver] dag_default_view](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-default-view) or even set it on dag level by set `default_view` argument

```python
from airflow import DAG

with DAG(
    default_view="graph",
    ...
) as dag:
    ...
```

```python
from airflow.decorators import dag

@dag(
    default_view="graph",
    ...
)
def awesome_dag():
    ...
```

**Comment by Liaomessi** [comment]
@potiuk I'm sorry for the late response. My previous view were one-sided. As you mentioned, I think it might be better to use a graph view when the dag graph does have a tree structure. When the intermediate tasks failed, we can use the tree view to find the subtree. But for some dag users who don't have complex dags, then the grid view might be enough.
@Taragolis Thanks, it works great for my problem.
